### DISCUSSION TITLE
Primary selection question

### DISCUSSION THREAD

**Discussion by seqizz** [erebe/greenclip#122] [Q&A] [answered]
Hi there :wave: 

Not maybe directly related to greenclip, but I realized a weird thing on my clipboard. Normally it is working as expected, but when I use some fancy web thing (like [this](https://www.w3schools.com/howto/howto_js_copy_clipboard.asp)), it uses the CLIPBOARD, and PRIMARY (maybe rightfully) stays intact.
My mind is not working like it though. Since normally "copy" is an action which I do via selecting, thus overriding PRIMARY too. Can I enforce any new CLIPBOARD action to override PRIMARY?

**Comment by erebe** [comment]
Is the config below config option not what you want ?
```
use_primary_selection_as_input = true
```

**Reply by seqizz** [reply]
Sadly it doesn't. Changed config to:
```
[greenclip]
  blacklisted_applications = []
  enable_image_support = true
  history_file = "/home/gurkan/.cache/greenclip.history"
  image_cache_directory = "/tmp/greenclip"
  max_history_length = 200
  max_selection_size_bytes = 0
  static_history = []
  trim_space_from_selection = true
  use_primary_selection_as_input = true
```
Restarted greenclip service. Selected and copied same thing to set it as base:

```
innodellix ~/.config ~> xclip -selection clipboard -o -t UTF8_STRING
/tmp/greenclip%
innodellix ~/.config ~> xclip -selection primary -o -t UTF8_STRING
/tmp/greenclip%
```
Went to the example link above, clicked copy. Here is the aftermath:
```
innodellix ~/.config ~> xclip -selection clipboard -o -t UTF8_STRING
Hello World%
innodellix ~/.config ~> xclip -selection primary -o -t UTF8_STRING
/tmp/greenclip%
```

**Comment by erebe** [comment, marked as answer]
Ha, I understand better !
Greenclip is not responsible for this. When you are using an application, it is the application that is the owner of the clipboard/selection. 
Greenclip only take ownership of the selection uniquely when you invoke it (greenclip print "xxx")

And sadly nowaday (especially app based on electron), they don't care about PRIMARY clipboard.

Don't know if I am clear, but greenclip is not responsible for this, it is the app that is in control of the clipboard.

**Reply by seqizz** [reply]
Yes, that's what I was guessing. But as a clipboard manager, which polls the clipboards' status periodically, I was expecting to find some kind of feature which basically says "if the value on PRIMARY is older than CLIPBOARD, override it with whatever CLIPBOARD has".
If I knew anything about Haskell, I'd send a patch now :grimacing: 
Thanks anyway :+1: I guess I need to train my muscle memory a bit.
